src/Client/Features/OfferMatching/List.razor

```diff
@@ -20,7 +20,7 @@
             </HeaderContent>
             <RowTemplate>
                 <MudTd DataLabel="Actions">
-                    <MudIconButton Icon="@Icons.Filled.Handshake" Color="Color.Primary" Variant="Variant.Outlined" Size="Size.Small" Class="ma-2" />
+                    <MudIconButton Icon="@Icons.Filled.Handshake" Color="Color.Primary" Variant="Variant.Outlined" Size="Size.Small" Class="ma-2" OnClick="(e) => MatchPlayerAsync(context.Id,context.Offer)"/>
```

Comment:
I don't think you need to pass the amount from here. Can you get the id and then determine the amount in the server?